// webapp/src/redux/stateSlice.jsx
// redux/adminSlice.js
import { createSlice } from '@reduxjs/toolkit';

// Load initial state from localStorage
const isLoggedInFromStorage = localStorage.getItem('isAdminLoggedIn') === 'true';
const userToken = localStorage.getItem('userToken');
const userData = JSON.parse(localStorage.getItem('userData') || 'null');

const initialState = {
  isAdminLoggedIn: isLoggedInFromStorage,
  user: {
    token: userToken,
    data: userData,
    isLoggedIn: !!userToken,
  },
};

const adminSlice = createSlice({
  name: 'admin',
  initialState,
  reducers: {
    loginAdmin: (state) => {
      state.isAdminLoggedIn = true;
      localStorage.setItem('isAdminLoggedIn', 'true');
    },
    logoutAdmin: (state) => {
      state.isAdminLoggedIn = false;
      localStorage.removeItem('isAdminLoggedIn');
    },
    loginUser: (state, action) => {
      state.user.token = action.payload.token;
      state.user.data = action.payload.user;
      state.user.isLoggedIn = true;
      localStorage.setItem('userToken', action.payload.token);
      localStorage.setItem('userData', JSON.stringify(action.payload.user));
    },
    logoutUser: (state) => {
      state.user.token = null;
      state.user.data = null;
      state.user.isLoggedIn = false;
      localStorage.removeItem('userToken');
      localStorage.removeItem('userData');
    },
  },
});

export const { loginAdmin, logoutAdmin, loginUser, logoutUser } = adminSlice.actions;
export default adminSlice.reducer;
