// webapp/src/components/Contact.jsx
import axios from 'axios';
import React, { useState } from 'react';
import Api_url from '../constant/constant';
import { MessageCircle, Instagram, Youtube, Facebook, Send, Linkedin, AtSign } from 'lucide-react';
import useSiteConfig from '../hooks/useSiteConfig';

const Contact = () => {
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    website: '',
    message: '',
  });
  const { socialLinks } = useSiteConfig();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const postdata = await axios.post(`${Api_url.BACKEND_URI}/api/contact`, formData);
      if (postdata.status === 201) {
        alert('Successfully sent your message. Thank you.');
      }
    } catch (error) {
      console.error('Error submitting form:', error);
    }
    setFormData({ name: '', email: '', website: '', message: '' });
  };

  const socialItems = [
    { key: 'whatsapp', label: 'WhatsApp', icon: <MessageCircle className="text-green-600" size={16} /> },
    { key: 'instagram', label: 'Instagram', icon: <Instagram className="text-pink-500" size={16} /> },
    { key: 'youtube', label: 'YouTube', icon: <Youtube className="text-red-600" size={16} /> },
    { key: 'facebook', label: 'Facebook', icon: <Facebook className="text-blue-600" size={16} /> },
    { key: 'telegram', label: 'Telegram', icon: <Send className="text-sky-500" size={16} /> },
    { key: 'linkedin', label: 'LinkedIn', icon: <Linkedin className="text-blue-700" size={16} /> },
    { key: 'x', label: 'X', icon: <AtSign className="text-black dark:text-white" size={16} /> },
  ].filter((item) => socialLinks?.[item.key]);

  return (
    <div className="min-h-screen bg-gray-50 py-12 px-4 sm:px-6 lg:px-8">
      <div className="max-w-4xl mx-auto">
        <h1 className="text-3xl font-bold text-center text-blue-700 mb-10">Contact Us</h1>

        {socialItems.length > 0 && (
          <div className="flex flex-wrap justify-center gap-4 mb-10">
            {socialItems.map((item) => (
              <a
                key={item.key}
                href={socialLinks[item.key]}
                target="_blank"
                rel="noopener noreferrer"
                className="flex items-center gap-2 rounded-lg bg-white px-3 py-2 text-sm font-medium shadow hover:shadow-md transition"
              >
                {item.icon}
                {item.label}
              </a>
            ))}
          </div>
        )}

        <form onSubmit={handleSubmit} className="bg-white shadow-lg rounded-lg p-8 space-y-6">
          <div>
            <label htmlFor="name" className="block font-medium text-gray-700 mb-1">Name</label>
            <input
              type="text"
              id="name"
              name="name"
              value={formData.name}
              onChange={handleChange}
              className="w-full border border-gray-300 px-4 py-2 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
              required
            />
          </div>

          <div>
            <label htmlFor="email" className="block font-medium text-gray-700 mb-1">Email</label>
            <input
              type="email"
              id="email"
              name="email"
              value={formData.email}
              onChange={handleChange}
              className="w-full border border-gray-300 px-4 py-2 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
              required
            />
          </div>

          <div>
            <label htmlFor="website" className="block font-medium text-gray-700 mb-1">Website (optional)</label>
            <input
              type="url"
              id="website"
              name="website"
              value={formData.website}
              onChange={handleChange}
              placeholder="https://example.com"
              className="w-full border border-gray-300 px-4 py-2 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-300"
            />
          </div>

          <div>
            <label htmlFor="message" className="block font-medium text-gray-700 mb-1">Message</label>
            <textarea
              id="message"
              name="message"
              value={formData.message}
              onChange={handleChange}
              rows="5"
              className="w-full border border-gray-300 px-4 py-2 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
              required
            ></textarea>
          </div>

          <div className="text-center">
            <button type="submit" className="bg-blue-600 text-white font-semibold py-2 px-6 rounded hover:bg-blue-700 transition duration-300">
              Send Message
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default Contact;
