// webapp/src/components/Footer.jsx
import axios from 'axios';
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Api_url from '../constant/constant';
import { useSelector } from 'react-redux';
import { MessageCircle, Instagram, Youtube, Facebook, Send, Linkedin, AtSign } from 'lucide-react';
import useSiteConfig from '../hooks/useSiteConfig';

const Footer = () => {
  const [email, setEmail] = useState('');
  const [message, setMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const showFooter = useSelector((state) => state.admin.isAdminLoggedIn);
  const { logoUrl, socialLinks } = useSiteConfig();
  const safeLogoUrl = (typeof logoUrl === 'string' && logoUrl.trim()) ? logoUrl : '/logo_.jpg';

  const handleSubscribe = async (e) => {
    e.preventDefault();
    setLoading(true);
    setMessage('');
    try {
      const res = await axios.post(`${Api_url.BACKEND_URI}/api/subscriber`, { email });
      if (res.status === 201 || res.status === 200) {
        setMessage('Subscribed successfully.');
        setEmail('');
      }
    } catch (error) {
      setMessage(error.response?.data?.message || 'Subscription failed. Please try again.');
    } finally {
      setLoading(false);
    }
  };

  if (showFooter) return null;

  const socialItems = [
    { key: 'whatsapp', icon: <MessageCircle size={18} />, color: 'text-green-500' },
    { key: 'instagram', icon: <Instagram size={18} />, color: 'text-pink-500' },
    { key: 'youtube', icon: <Youtube size={18} />, color: 'text-red-500' },
    { key: 'facebook', icon: <Facebook size={18} />, color: 'text-blue-500' },
    { key: 'telegram', icon: <Send size={18} />, color: 'text-sky-500' },
    { key: 'linkedin', icon: <Linkedin size={18} />, color: 'text-blue-600' },
    { key: 'x', icon: <AtSign size={18} />, color: 'text-slate-800 dark:text-white' },
  ].filter((item) => socialLinks?.[item.key]);

  return (
    <footer className="relative mt-16 border-t border-blue-100 bg-gradient-to-br from-slate-950 via-blue-950 to-slate-900 text-slate-100">
      <div className="pointer-events-none absolute inset-0 bg-[radial-gradient(circle_at_top_right,rgba(96,165,250,0.2),transparent_45%)]" />
      <div className="relative mx-auto max-w-7xl px-4 py-12 sm:px-6 lg:px-8">
        <div className="grid gap-10 md:grid-cols-2 lg:grid-cols-4">
          <section>
            <div className="mb-4 flex items-center gap-3">
              <img
                src={safeLogoUrl}
                alt="Scholarship for Studneet"
                className="h-14 w-14 rounded-full border border-blue-300/40 object-cover"
                onError={(e) => { e.currentTarget.onerror = null; e.currentTarget.src = '/logo_.jpg'; }}
              />
              <div>
                <p className="text-lg font-semibold tracking-wide">Scholarship For Studneet</p>
                <p className="text-xs text-blue-200">Global opportunities for students</p>
              </div>
            </div>
            <p className="text-sm leading-relaxed text-slate-300">
              Discover updated scholarships, internships, and leadership programs in one place.
              Apply smarter with clear deadlines and trusted links.
            </p>

            {socialItems.length > 0 && (
              <div className="mt-4 flex flex-wrap gap-3 text-xl">
                {socialItems.map((item) => (
                  <a
                    key={item.key}
                    href={socialLinks[item.key]}
                    target="_blank"
                    rel="noopener noreferrer"
                    className={`rounded-full bg-white/90 p-2 transition hover:scale-110 ${item.color}`}
                  >
                    {item.icon}
                  </a>
                ))}
              </div>
            )}
          </section>

          <section>
            <h3 className="mb-4 text-base font-semibold text-white">Quick Links</h3>
            <ul className="space-y-2 text-sm text-slate-300">
              <li><Link to="/" className="transition hover:text-blue-300">Home</Link></li>
              <li><Link to="/about" className="transition hover:text-blue-300">About</Link></li>
              <li><Link to="/contact" className="transition hover:text-blue-300">Contact</Link></li>
              <li><Link to="/youtube-videos" className="transition hover:text-blue-300">YouTube</Link></li>
              <li><Link to="/faq" className="transition hover:text-blue-300">FAQ</Link></li>
              <li><Link to="/competition" className="transition hover:text-blue-300">Competition</Link></li>
            </ul>
          </section>

          <section>
            <h3 className="mb-4 text-base font-semibold text-white">Explore Scholarships</h3>
            <ul className="grid grid-cols-1 gap-2 text-sm text-slate-300 sm:grid-cols-2 lg:grid-cols-1">
              <li><Link to="/european-scholarship" className="transition hover:text-blue-300">European</Link></li>
              <li><Link to="/asian_scholarship" className="transition hover:text-blue-300">Asian</Link></li>
              <li><Link to="/african-scholarship" className="transition hover:text-blue-300">African</Link></li>
              <li><Link to="/american-scholarship" className="transition hover:text-blue-300">American</Link></li>
              <li><Link to="/australian-scholarship" className="transition hover:text-blue-300">Australian</Link></li>
              <li><Link to="/south-american-scholarship" className="transition hover:text-blue-300">South American</Link></li>
              <li><Link to="/middle_eastscholarship" className="transition hover:text-blue-300">Middle East</Link></li>
              <li><Link to="/chinese_govt_scholarship" className="transition hover:text-blue-300">Chinese Govt</Link></li>
            </ul>
          </section>

          <section>
            <h3 className="mb-4 text-base font-semibold text-white">Get Weekly Updates</h3>
            <form onSubmit={handleSubscribe} className="space-y-3">
              <input
                type="email"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                placeholder="Enter your email"
                required
                className="w-full rounded-lg border border-blue-300/30 bg-white/95 px-3 py-2 text-sm text-slate-900 outline-none ring-blue-400 transition focus:ring-2"
              />
              <button
                type="submit"
                disabled={loading}
                className="w-full rounded-lg bg-gradient-to-r from-blue-500 to-indigo-500 px-4 py-2 text-sm font-semibold text-white transition hover:from-blue-600 hover:to-indigo-600 disabled:cursor-not-allowed disabled:opacity-70"
              >
                {loading ? 'Subscribing...' : 'Subscribe'}
              </button>
            </form>
            {message && (
              <p className="mt-3 text-xs font-medium text-blue-200">{message}</p>
            )}
          </section>
        </div>

        <div className="mt-10 flex flex-col items-start justify-between gap-3 border-t border-blue-300/20 pt-5 text-xs text-slate-400 sm:flex-row sm:items-center">
          <p>© 2026 Scholarship For Studneet. All rights reserved.</p>
          <p>Built for students, educators, and global applicants.</p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
