// webapp/src/admin/Add_schoarship.jsx
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import apiClient from '../services/apiClient';

const categories = ['summer_programe', 'famouse_scholarship', 'compition', 'Undergraduate', 'Master', 'PhD', 'highschool', 'exchangeprograme'];
const regions = [
  'south america',
  'Africa',
  'Asia',
  'Europe',
  'North America',
  'South America',
  'Australia',
  'Middle East',
  'Oceania',
  'Global',
  'american',
  'Russia',
];

const Add_schoarship = ({ onSubmit }) => {
  const navigate = useNavigate();
  const [form, setForm] = useState({
    id: '',
    name: '',
    imageFile: null,
    description: '',
    category: [],
    benefits: '',
    eligibilityCriteria: '',
    amount: '',
    deadline: '',
    region: '',
    country: '',
    officialLink: '',
    document: '',
    requiredDocuments: '',
    hostUniversity: '',
    howToApply: '',
  });
  const [error, setError] = useState('');
  const [success, setSuccess] = useState('');
  const [loading, setLoading] = useState(false);

  const handleChange = (e) => {
    const { name, value, type, checked, files } = e.target;
    if (name === 'category') {
      setForm((prev) => ({
        ...prev,
        category: checked
          ? [...prev.category, value]
          : prev.category.filter((cat) => cat !== value),
      }));
      return;
    }

    if (name === 'imageFile') {
      setForm((prev) => ({ ...prev, imageFile: files?.[0] || null }));
      return;
    }

    setForm((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setError('');
    setSuccess('');
    setLoading(true);

    try {
      const payload = new FormData();
      payload.append('id', form.id);
      payload.append('name', form.name);
      payload.append('description', form.description);
      form.category.forEach((item) => payload.append('category', item));
      payload.append('benefits', form.benefits);
      payload.append('eligibilityCriteria', form.eligibilityCriteria);
      payload.append('amount', String(Number(form.amount || 0)));
      payload.append('deadline', form.deadline);
      payload.append('region', form.region);
      payload.append('country', form.country);
      payload.append('officialLink', form.officialLink);
      payload.append('document', form.document);
      payload.append('requiredDocuments', form.requiredDocuments);
      payload.append('hostUniversity', form.hostUniversity);
      payload.append('howToApply', form.howToApply);
      if (form.imageFile) {
        payload.append('imageFile', form.imageFile);
      }

      await apiClient.post('/save/scholarshipdata', payload, {
        headers: { 'Content-Type': 'multipart/form-data' },
      });

      setSuccess('Scholarship saved successfully.');
      setForm({
        id: '',
        name: '',
        imageFile: null,
        description: '',
        category: [],
        benefits: '',
        eligibilityCriteria: '',
        amount: '',
        deadline: '',
        region: '',
        country: '',
        officialLink: '',
        document: '',
        requiredDocuments: '',
        hostUniversity: '',
        howToApply: '',
      });
      if (onSubmit) onSubmit();
    } catch (err) {
      setError(err.response?.data?.message || 'Error saving scholarship data.');
    } finally {
      setLoading(false);
    }
  };

  return (
    <div className="space-y-4">
      <button className="rounded-lg bg-slate-700 px-4 py-2 text-sm font-semibold text-white" onClick={() => navigate(-1)}>
        Go Back
      </button>

      <form className="mx-auto max-w-3xl space-y-4 rounded-xl border border-slate-200 bg-white p-4 shadow-sm sm:p-6" onSubmit={handleSubmit}>
        <h2 className="text-xl font-bold text-slate-800 sm:text-2xl">Add Scholarship</h2>

        {error && <div className="rounded border border-red-300 bg-red-50 p-2 text-sm text-red-700">{error}</div>}
        {success && <div className="rounded border border-green-300 bg-green-50 p-2 text-sm text-green-700">{success}</div>}

        <div className="grid gap-4 sm:grid-cols-2">
          <div>
            <label className="mb-1 block text-sm font-medium">ID</label>
            <input type="text" name="id" value={form.id} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
          </div>
          <div>
            <label className="mb-1 block text-sm font-medium">Name</label>
            <input type="text" name="name" value={form.name} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
          </div>
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">Scholarship Image Upload</label>
          <input type="file" name="imageFile" accept="image/*,.bmp" onChange={handleChange} className="w-full rounded border px-3 py-2" required />
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">Description</label>
          <textarea name="description" value={form.description} onChange={handleChange} rows={3} className="w-full rounded border px-3 py-2" required />
        </div>

        <div>
          <span className="mb-1 block text-sm font-medium">Category</span>
          <div className="grid gap-2 sm:grid-cols-2">
            {categories.map((cat) => (
              <label key={cat} className="flex items-center gap-2 text-sm">
                <input
                  type="checkbox"
                  name="category"
                  value={cat}
                  checked={form.category.includes(cat)}
                  onChange={handleChange}
                />
                <span>{cat}</span>
              </label>
            ))}
          </div>
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">Benefits</label>
          <textarea name="benefits" value={form.benefits} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">Eligibility Criteria</label>
          <textarea name="eligibilityCriteria" value={form.eligibilityCriteria} onChange={handleChange} rows={3} className="w-full rounded border px-3 py-2" required />
        </div>

        <div className="grid gap-4 sm:grid-cols-2">
          <div>
            <label className="mb-1 block text-sm font-medium">Amount</label>
            <input type="number" name="amount" value={form.amount} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
          </div>
          <div>
            <label className="mb-1 block text-sm font-medium">Deadline</label>
            <input type="date" name="deadline" value={form.deadline} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
          </div>
        </div>

        <div className="grid gap-4 sm:grid-cols-2">
          <div>
            <label className="mb-1 block text-sm font-medium">Region</label>
            <select name="region" value={form.region} onChange={handleChange} className="w-full rounded border px-3 py-2" required>
              <option value="">Select Region</option>
              {regions.map((region) => (
                <option key={region} value={region}>{region}</option>
              ))}
            </select>
          </div>
          <div>
            <label className="mb-1 block text-sm font-medium">Country</label>
            <input type="text" name="country" value={form.country} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
          </div>
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">Official Link</label>
          <input type="url" name="officialLink" value={form.officialLink} onChange={handleChange} className="w-full rounded border px-3 py-2" required />
        </div>

        <div className="grid gap-4 sm:grid-cols-2">
          <div>
            <label className="mb-1 block text-sm font-medium">Document (optional)</label>
            <input type="text" name="document" value={form.document} onChange={handleChange} className="w-full rounded border px-3 py-2" />
          </div>
          <div>
            <label className="mb-1 block text-sm font-medium">Required Documents (Russian scholarship)</label>
            <input
              type="text"
              name="requiredDocuments"
              value={form.requiredDocuments}
              onChange={handleChange}
              placeholder="Passport, transcript, language certificate..."
              className="w-full rounded border px-3 py-2"
            />
          </div>
          <div>
            <label className="mb-1 block text-sm font-medium">Host University (optional)</label>
            <input type="text" name="hostUniversity" value={form.hostUniversity} onChange={handleChange} className="w-full rounded border px-3 py-2" />
          </div>
        </div>

        <div>
          <label className="mb-1 block text-sm font-medium">How to Apply (optional)</label>
          <textarea name="howToApply" value={form.howToApply} onChange={handleChange} rows={3} className="w-full rounded border px-3 py-2" />
        </div>

        <button type="submit" disabled={loading} className="w-full rounded bg-blue-600 py-2 font-semibold text-white hover:bg-blue-700 disabled:opacity-70">
          {loading ? 'Submitting...' : 'Submit'}
        </button>
      </form>
    </div>
  );
};

export default Add_schoarship;
