// webapp/src/admin/Add_internship.jsx
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import apiClient from '../services/apiClient';

const initialState = {
  internship_id: '',
  custome_message: '',
  name: '',
  imageFile: null,
  description: '',
  hosted_country: '',
  document: '',
  eligabality_criteria: '',
  officialLink: '',
  duration: '',
  benifits: '',
  application_process: '',
  deadline: '',
};

const Add_internship = () => {
  const navigate = useNavigate();
  const [form, setForm] = useState(initialState);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState('');
  const [success, setSuccess] = useState('');
  const showPage = useSelector((state) => state.admin.isAdminLoggedIn);

  const handleChange = (e) => {
    const { name, value, files } = e.target;
    if (name === 'imageFile') {
      setForm((prev) => ({ ...prev, imageFile: files?.[0] || null }));
      return;
    }
    setForm((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setLoading(true);
    setError('');
    setSuccess('');
    try {
      const payload = new FormData();
      Object.entries(form).forEach(([key, value]) => {
        if (key === 'imageFile') return;
        payload.append(key, value ?? '');
      });
      if (form.imageFile) payload.append('imageFile', form.imageFile);

      await apiClient.post('/admin/saveInternship', payload, {
        headers: { 'Content-Type': 'multipart/form-data' },
      });
      setSuccess('Internship added successfully!');
      setForm(initialState);
      navigate('/admin/admin_dashboard');
    } catch (err) {
      setError('Error occurred: ' + (err.response?.data?.message || err.message));
    }
    setLoading(false);
  };

  if (!showPage) return null;

  return (
    <div className="space-y-4">
      <button className="rounded bg-slate-800 px-4 py-2 text-white" onClick={() => navigate(-1)}>Back</button>
      <div className="mx-auto max-w-3xl rounded-xl border border-slate-200 bg-white p-4 shadow-sm sm:p-6">
        <h2 className="mb-4 text-2xl font-bold">Add Internship</h2>
        <form onSubmit={handleSubmit} className="grid gap-4">
          <input type="text" name="internship_id" value={form.internship_id} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Internship ID" required />
          <input type="text" name="custome_message" value={form.custome_message} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Custom Message" required />
          <input type="text" name="name" value={form.name} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Internship Name" required />
          <div>
            <label className="mb-1 block text-sm font-medium">Internship Image Upload</label>
            <input type="file" name="imageFile" accept="image/*,.bmp" onChange={handleChange} className="w-full border rounded px-3 py-2" required />
          </div>
          <textarea name="description" value={form.description} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Description" required />
          <input type="text" name="hosted_country" value={form.hosted_country} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Hosted Country" required />
          <input type="text" name="document" value={form.document} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Document" required />
          <textarea name="eligabality_criteria" value={form.eligabality_criteria} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Eligibility" required />
          <input type="url" name="officialLink" value={form.officialLink} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Official Link" required />
          <input type="text" name="duration" value={form.duration} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Duration" required />
          <textarea name="benifits" value={form.benifits} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Benefits" required />
          <textarea name="application_process" value={form.application_process} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Application Process" required />
          <input type="date" name="deadline" value={form.deadline} onChange={handleChange} className="w-full border rounded px-3 py-2" required />
          {error && <div className="text-red-600 text-sm">{error}</div>}
          {success && <div className="text-green-600 text-sm">{success}</div>}
          <button type="submit" className="w-full bg-blue-600 text-white py-2 rounded hover:bg-blue-700" disabled={loading}>
            {loading ? 'Submitting...' : 'Add Internship'}
          </button>
        </form>
      </div>
    </div>
  );
};

export default Add_internship;
