// webapp/src/admin/Add_leadership.jsx
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import apiClient from '../services/apiClient';

const initialState = {
  id: '',
  programTitle: '',
  imageFile: null,
  hostCountry: '',
  hostOrganization: '',
  programLocation: '',
  duration: '',
  benefits: '',
  eligibility: '',
  howToApply: '',
  documentsRequired: '',
  deadline: '',
  description: '',
  officialLink: '',
};

const Add_leadership = () => {
  const navigate = useNavigate();
  const showLeadershpprogram = useSelector((state) => state.admin.isAdminLoggedIn);
  const [form, setForm] = useState(initialState);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState('');
  const [success, setSuccess] = useState('');

  const handleChange = (e) => {
    const { name, value, files } = e.target;
    if (name === 'imageFile') {
      setForm((prev) => ({ ...prev, imageFile: files?.[0] || null }));
      return;
    }
    setForm((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setLoading(true);
    setError('');
    setSuccess('');
    try {
      const payload = new FormData();
      Object.entries(form).forEach(([key, value]) => {
        if (key === 'imageFile') return;
        payload.append(key, value ?? '');
      });
      if (form.imageFile) payload.append('imageFile', form.imageFile);

      await apiClient.post('/add/leader_data', payload, {
        headers: { 'Content-Type': 'multipart/form-data' },
      });
      setSuccess('Leadership program added successfully!');
      setForm(initialState);
    } catch (err) {
      setError('Error occurred: ' + (err.response?.data?.message || err.message));
    }
    setLoading(false);
  };

  if (!showLeadershpprogram) return <p className="text-sm text-slate-600">Login first.</p>;

  return (
    <div className="space-y-4">
      <button className="rounded bg-red-500 px-4 py-2 text-white" onClick={() => navigate(-1)}>Back</button>
      <div className="mx-auto max-w-3xl rounded-xl border border-slate-200 bg-white p-4 shadow-sm sm:p-6">
        <h2 className="mb-4 text-2xl font-bold">Add Leadership Program</h2>
        <form onSubmit={handleSubmit} className="grid gap-4">
          <input type="text" name="id" value={form.id} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Program id" required />
          <input type="text" name="programTitle" value={form.programTitle} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Program title" required />
          <div>
            <label className="mb-1 block text-sm font-medium">Program Image Upload</label>
            <input type="file" name="imageFile" accept="image/*,.bmp" onChange={handleChange} className="w-full border rounded px-3 py-2" required />
          </div>
          <input type="text" name="hostCountry" value={form.hostCountry} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Host Country" required />
          <input type="text" name="hostOrganization" value={form.hostOrganization} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Host Organization" required />
          <input type="text" name="programLocation" value={form.programLocation} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Program Location" required />
          <input type="text" name="duration" value={form.duration} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Duration" required />
          <textarea name="benefits" value={form.benefits} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Benefits" required />
          <textarea name="eligibility" value={form.eligibility} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Eligibility" required />
          <textarea name="howToApply" value={form.howToApply} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="How to apply" required />
          <textarea name="documentsRequired" value={form.documentsRequired} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Documents required" required />
          <input type="date" name="deadline" value={form.deadline} onChange={handleChange} className="w-full border rounded px-3 py-2" required />
          <textarea name="description" value={form.description} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Description" required />
          <input type="url" name="officialLink" value={form.officialLink} onChange={handleChange} className="w-full border rounded px-3 py-2" placeholder="Official Link" />
          {error && <div className="text-red-600 text-sm">{error}</div>}
          {success && <div className="text-green-600 text-sm">{success}</div>}
          <button type="submit" className="w-full bg-blue-600 text-white py-2 rounded hover:bg-blue-700" disabled={loading}>
            {loading ? 'Submitting...' : 'Add Leadership Program'}
          </button>
        </form>
      </div>
    </div>
  );
};

export default Add_leadership;
